index.js

```diff
@@ -23,13 +23,22 @@ module.exports.pitch = function (request) {
 		"// Hot Module Replacement",
 		"if(module.hot) {",
 		"	// When the styles change, update the <style> tags",
-		"	if(!content.locals) {",
-		"		module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
-		"			var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
-		"			if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
-		"			update(newContent);",
-		"		});",
-		"	}",
+		"	module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
+		"		var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
+		"		if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
+		"		var locals = (function(a, b) {",
+		"			var key, idx = 0;",
+		"			for(key in a) {",
+		"				if(!b || a[key] !== b[key]) return false;",
+		"				idx++;",
+		"			}",
+		"			for(key in b) idx--;",
+		"			return idx === 0;",
+		"		}(content.locals, newContent.locals));",
+		"		// This error is caught and not shown and causes a full reload.",
+		"		if(!locals) throw new Error('Aborting CSS HMR due to changed css-modules locals.');",
```

Comment:
Let's remove comment from source code (move as comment before this line), it is increase bundle in dev mode, Good work, thanks!

Reply:
I see what you mean, but wouldn't that be inconsistent with all the other comments that are part of the output source code?

Reply:
You're right, but the fact is that this code needs a few refactorings anyways in the future :)  @evilebottnawi is a good idea and **I will do it** for all the source comments before cutting a release. @evilebottnawi anything else of concern otherwise I would land this now and take care of the rest ?

Reply:
@lydell oh yes, let's fix it in next PR

Reply:
@michael-ciniawsky merge as is and fix it next PR